ecoli/analysis/multivariant/cell_mass.py

```diff
@@ -0,0 +1,162 @@
+"""
+Plot absolue / normalized cell mass over time for multivariant simulation in vEcoli, and:
+1. each variant has its own plot;
+2. at each subplot, time is divided by generation id;
+
+It can also be used at multigeneration analysis.
+"""
+
+import os
+from typing import Any
+import altair as alt
+import polars as pl
+import pandas as pd
+from duckdb import DuckDBPyConnection
+
+
+def plot(
+    params: dict[str, Any],
+    conn: DuckDBPyConnection,
+    history_sql: str,
+    config_sql: str,
+    success_sql: str,
+    sim_data_dict: dict[str, dict[int, str]],
+    validation_data_paths: list[str],
+    outdir: str,
+    variant_metadata: dict[str, dict[int, Any]],
+    variant_names: dict[str, str],
+):
+    # Load data with required columns
+    required_columns = [
+        "time",
+        "variant",
+        "generation",
+        "agent_id",
+        "listeners__mass__dry_mass",
+    ]
+
+    sql = f"""
+    SELECT {", ".join(required_columns)}
+    FROM ({history_sql})
+    WHERE agent_id = 0
+    ORDER BY variant, generation, time
+    """
+
+    df = conn.sql(sql).pl()
+
+    # Process time and mass data
+    df = df.with_columns(
+        [
+            (pl.col("time") / 60).alias("time_min"),
+        ]
+    )
+
+    # Calculate initial mass for each generation for normalization
+    generation_stats = df.group_by(["variant", "generation"]).agg(
```

Comment:
This is not quite right because there are weird cases where the cell mass will go down slightly at the start of a generation. There are ways to get the dry mass at the first time step (probably using a join), but for this plot, I'd recommend just using the `listeners__mass__dry_mass_fold_change` column.